amlb/resources.py

```diff
@@ -77,13 +120,18 @@ def _seed(self):
         else:
             return self.config.seed
 
-    def framework_definition(self, name):
+    def framework_definition(self, name, tag=None):
         """
         :param name:
         :return: name of the framework as defined in the frameworks definition file
         """
         lname = name.lower()
-        framework = next((f for n, f in self._frameworks if n.lower() == lname), None)
+        if tag is None:
+            tag = default_tag
+        if tag not in self._frameworks:
+            raise ValueError("Incorrect tag `{}`: only those among {} are allowed.".format(tag, self.config.frameworks.tags))
```

Comment:
Since we're Py3.6+, is there a reason not to use `f-string` here?

Reply:
matter of taste I guess but I try not to use `f-string` if the expression is too long, I think it makes the string noisy.
For example here:
- f"Incorrect tag `{tag}`: only those among {tags} are allowed."    -> I like, the string is easy to read.
- f"Incorrect tag `{tag}`: only those among {self.config.frameworks.tags} are allowed." -> I don't like, the placeholder represents too much of the sentence and I find it noisy in the context.

Reply:
Fair enough! For me the latter is still preferred so I don't have to go back and forth to see what goes where, but to each their own.